api/envoy/type/matcher/v3/filter_state.proto

```diff
@@ -25,5 +27,12 @@ message FilterStateMatcher {
 
     // Matches the filter state object as a string value.
     StringMatcher string_match = 2;
+
+    // Matches the filter state object as a ip Instance.
+    AddressMatcher address_match = 3;
   }
 }
+
+message AddressMatcher {
+  repeated xds.core.v3.CidrRange ranges = 1;
+}
```

Comment:
Sorry, I didn't make it more clear. I mean you could create a new proto file (`address.proto`) at the matcher/v3. And then, add the `AddressMatcher` there.